src/field_theory/krull_topology.lean

```diff
@@ -287,3 +287,71 @@ begin
 end
 
 end totally_disconnected
+
+
+section finite
+
+variables {K E L : Type*} [field K] [field E] [algebra K E] [field L] [algebra K L]
+
+/-- A technical finiteness result. -/
+noncomputable def fintype.subtype_prod {E : Type*} {X : set E} (hX : X.finite) {L : Type*}
+  (F : E → multiset L) : fintype (Π x : X, {l : L // l ∈ F x}) :=
+by { classical, letI : fintype X := set.finite.fintype hX, exact pi.fintype}
+
+variable (K)
+
+/-- Function taking `e ∈ E` to the multiset of roots of its minimal polynomial in `L` -/
+noncomputable def roots_of_min_poly_multiset : E → multiset L :=
+λ e, ((minpoly K e).map (algebra_map K L)).roots
```

Comment:
```suggestion
noncomputable def roots_of_min_poly_multiset (e : E) : multiset L :=
((minpoly K e).map (algebra_map K L)).roots
```
I think I would just call this `minpoly_roots`. Also, this seems quite general, so it should probably move to another file. Do you really need it, or can you just inline the definition? It seems that you only use it once, and immediately unfold the definition in the proof below it.